zerver/tests/test_auth_backends.py

```diff
@@ -759,6 +777,299 @@ def test_github_complete_when_base_exc_is_raised(self) -> None:
 
         utils.BACKENDS = settings.AUTHENTICATION_BACKENDS
 
+@override_settings(AUTHENTICATION_BACKENDS=('zproject.backends.SAMLAuthBackend',),
```

Comment:
We'll probably want to rewrite this test to work how the new GitHubAuthBackendTest works (which is a much less mock-heavy, end-to-end test); basically we'd need to write the equivalent of `github_oauth2_test`.  The `httpretty` parts are based on how `python-social-auth` tests the backends in its own codebase; that's a useful resource that you'll want to look at for inspiration.